### ISSUE TITLE
NRE in ExceptionFrame.cs (or maybe a mono bug)

### ISSUE THREAD

**Issue by razzmatazz** [getsentry/raven-csharp#101]
Hi,

I have an issue where the following stack trace is emitted:
```
[ERROR] System.NullReferenceException: Object reference not set to an instance of an object
  at SharpRaven.Data.ExceptionFrame..ctor (System.Diagnostics.StackFrame frame) [0x00034] in /Users/bob/src/raven-csharp/src/app/SharpRaven/Data/ExceptionFrame.cs:64 
  at SharpRaven.Data.SentryStacktrace..ctor (System.Exception exception) [0x00043] in /Users/bob/src/raven-csharp/src/app/SharpRaven/Data/SentryStacktrace.cs:66 
  at SharpRaven.Data.SentryException..ctor (System.Exception exception) [0x00042] in /Users/bob/src/raven-csharp/src/app/SharpRaven/Data/SentryException.cs:60 
  at SharpRaven.Data.JsonPacket..ctor (System.String project, System.Exception exception) [0x00092] in /Users/bob/src/raven-csharp/src/app/SharpRaven/Data/JsonPacket.cs:77 
  at SharpRaven.Data.JsonPacketFactory.Create (System.String project, System.Exception exception, SharpRaven.Data.SentryMessage message, ErrorLevel level, IDictionary`2 tags, System.Object extra) [0x00000] in /Users/bob/src/raven-csharp/src/app/SharpRaven/Data/JsonPacketFactory.cs:101 
  at SharpRaven.RavenClient.CaptureException (System.Exception exception, SharpRaven.Data.SentryMessage message, ErrorLevel level, IDictionary`2 tags, System.Object extra) [0x00000] in /Users/bob/src/raven-csharp/src/app/SharpRaven/RavenClient.cs:153 
```

The problem is that code on line 64
```
62            var method = frame.GetMethod();
63            Filename = frame.GetFileName();
64            Module = (method.DeclaringType != null) ? method.DeclaringType.FullName : null;
```
is expecting for the `method` variable to be non-null as it is dereferencing `method.DeclaringType` immediately. In my case, however, this assumption fails for some reason and then I get a NRE.

The platform is mono 4.2.1 x86, raven-csharp @ master (commit 74a771068ef660558c3a1dd6ff75be21c1bcc45c).

I *think* it might be a mono bug, but not so sure. MS documentation on `StackFrame.GetMethod` does not apparently say anything on the possibility for the object returned to be null. Thus I assume this is a mono bug?

Should there be a workaround in the meantime for this? What is the policy for commits agains bugs in the underlying platform?

**Comment by asbjornu** [comment]
@razzmatazz If you can come up with a neat default value in place of `null`, I'd be more than happy to accept a pull request. I don't believe a policy for bugs in the underlying platform exist, but if a simple `??` or `?:` can fix an issue for you, I don't really care where the bug originates from. :)

**Comment by razzmatazz** [comment]
Good :) One moment, will try to add a test there and will post the PR later.

The actual issue appears to be that mono is returning a `null` for this stack frame:
```
  at (wrapper dynamic-method) System.Object:lambda_method (System.Runtime.CompilerServices.Closure,object,object)
```

So it is probably not a bug per-se, just a platform issue.

One question I have is: what is a "neat default value". I am thinking about using `(unknown)` but not sure if that is how other platforms do on sentry (so it would look consistent).

This is how it looks with `(unknown)` as placeholder for unknown/null frame:

<img width="969" alt="bildschirmfoto 2016-01-22 um 13 02 20" src="https://cloud.githubusercontent.com/assets/563556/12509068/a8e26cde-c108-11e5-91bc-808821913c3a.png">

**Comment by razzmatazz** [comment]
btw, `frame.ToString()` returns
```
<unknown method> at <unknown offset> in file:line:column <filename unknown>:0:0
```
for this frame so that is of no use either – I was thinking about using that for setting the `.Function` attribute.

**Comment by asbjornu** [comment]
Fixed in 9819624226e853195d75282a2ea304b4bcd2ed78.
